### ISSUE TITLE
When amending previous commit, only use staged files

### ISSUE THREAD

**Issue by Drllap** [divmain/GitSavvy#341]
When using [m] in STATUS window, all of the unstaged files are added to staged before amending. I think you should be able to add files to staged using [s] as you would in a normal commit and then those files would be used when amending

**Comment by divmain** [comment]
Could you provide additional information about the conditions under which this occurred?  I amend pretty regularly, and have never run into the issue you describe.  Amending should only add whatever is staged to the previous commit.

**Comment by Drllap** [comment]
Sorry for the late response.
So I tested this a bit more. The thing is I've got "show_commit_diff": true in my GitSavvy settings file.
When I amend, either from STATUS VIEW or via "git: amend previous commit" in the command palette, the diff for all staged and unstaged files (not untracked though) appears below the commit message region. This is what was confusing me, I thought if I proceeded with the amend, all of the diff would be added to the previous committed commit.

> Amending should only add whatever is staged to the previous commit.

This is not the behaviour I would expect nor what I have observed.
If I commit changes to "staged_file.h". Then change it, and do amend without staging the file before, the only changes that I can make to the commit, is to the commit message.
To change the actual commit, I have to stage the file again and then to the amend, as it should be.

I'm using
Sublime Text 3, build 3083
Windows 7 and 8.1
git version 2.7.0.windows.1

tl;dr
The amend itself is working as intended, but when you have show_commit_diff: true, the diff for all files appears in the Commit View when amending

**Comment by alextegelid** [comment]
> The amend itself is working as intended, but when you have show_commit_diff: true, the diff for all files appears in the Commit View when amending

:+1: I've noticed this as well, but haven't had the time to report it (and then forgot about it)

**Comment by divmain** [comment]
This should be good to go.
